frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -870,7 +885,15 @@ final class BloopBspServices(
   def buildTargets(
       request: bsp.WorkspaceBuildTargetsRequest
   ): BspEndpointResponse[bsp.WorkspaceBuildTargetsResult] = {
-    ifInitialized(None) { (state: State, logger: BspServerLogger) =>
+    // need a separate block so so that state is refreshed after regenerating project data
+    val refreshTask = ifInitialized(None) { (state: State, logger: BspServerLogger) =>
+      state.client.refreshProjectsCommand.foreach { command =>
```

Comment:
We should be running this within a `Task` so that if it ever throws (I know it doesn't now) we catch it within the `Task` and we can report the error to the client. Also, what's the contract if this command fails? I see you're right now ignoring the error, shouldn't the client know that this side effect failed?

Reply:
now the behavior is defined by Forker.run logic which also returns a Task